### ISSUE TITLE
Handling conda labelled channels

### ISSUE THREAD

**Issue by jsignell** [badges/shields#2955]
:clipboard: **Description**

When a conda channel has labels, the latest_version doesn't need to match the last version in the versions list. I propose that there be a new badge for conda like:

https://img.shields.io/conda/vln/:channel/:package.svg 

channel | v0.4.0a0

![](https://img.shields.io/badge/pyviz-v0.4.0a0-blue.svg)

:link: **Data**

Same public Anaconda Cloud API that is already used for the other conda badges: 
https://api.anaconda.org/package/{channel}/{package}
example: https://api.anaconda.org/package/pyviz/panel 

The part to use would be the last item in versions: `package.versions[package.versions.length - 1];`

:microphone: **Motivation**

I am building a [status dashboard](http://status.pyviz.org) for the [pyviz project](http://pyviz.org) and we publish dev releases to pyviz/label/dev.

**Comment by chris48s** [comment]
Can you explain the concept of labels slightly more. Is this basically the same as an "include pre-releases" flag? i.e: `latest_version` must be a stable release, but the most recent release in `versions` can be a pre-release, or is this a different concept? i.e: could `latest_version` independently be a pre-release?

**Comment by jsignell** [comment]
I looked back at the server coda and it seems that the `latest_version` has to be from the 'main' label. So when you upload a package to Anaconda Cloud you can set a label flag and if you do, the version you are uploading **will not** show up as `latest_version` but it **will** show up in the `versions` list. I think it would be fine to have the name of the badge be pre-release or something and it would be on the user to know that in the case of Anaconda Cloud that means using a labelled channel.

**Comment by paulmelnikow** [comment]
So the default label is `main` and that's what our badge checks now – is that right?

If I'm understanding correctly, this is similar to an npm tag. We have a badge that lets you pick a different tag like `next` or `canary`. In that case the badge URL specifies the tag, and the tag appears on the badge by default (though that can be customized).

How about we do the same here? (prereleases and unstable versions in orange)

https://img.shields.io/conda/vn/:channel/:package/:label.svg

![](https://img.shields.io/badge/pyviz@dev-0.4.0.a0-orange.svg)

**Comment by paulmelnikow** [comment]
It's all in one request though, right? That could be done with `.filter()`. It doesn't sound too bad :)

**Comment by jsignell** [comment]
Yep all in one request - the same request as the other conda methods.
